src/stringifyPrepare.js

```diff
@@ -83,7 +88,7 @@ function pruneArray(array, path, serializationSymbol, assignments) {
 function pruneObject(obj, path, serializationSymbol, assignments) {
     var clone = {};
 
-    if (obj.toJSON && obj.constructor != Date) {
+    if (obj.toJSON && obj.constructor != Date && obj.constructor != URL && obj.constructor != URLSearchParams) {
```

Comment:
`URL` and `URLSearchParams` do not implement `toJSON`, so this change should not be needed.

Reply:
Apparently URL does implement [toJSON](https://developer.mozilla.org/en-US/docs/Web/API/URL/toJSON) but I will remove it for URLSearchParams